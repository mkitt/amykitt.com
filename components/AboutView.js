// @flow
import React from 'react'
import Markdown from './Markdown'
import View from './View'
import type { About } from '../types/app.js.flow'

type Props = {
  about: About,
}

export default ({ about, ...props }: Props) => (
  <View {...props} >
    <Markdown
      source={about.body}
    />
  </View>
)
