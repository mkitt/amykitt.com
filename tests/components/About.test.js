import React from 'react'
import renderer from 'react-test-renderer'
import AboutView from '../../components/AboutView'

describe('AboutView', () => {
  let component

  beforeEach(() => {
    const about = {
      body: 'About _body_',
      id: '1',
      title: 'About title',
    }

    component = <AboutView about={about} />
  })

  it('shows the correct snapshot for the About component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
