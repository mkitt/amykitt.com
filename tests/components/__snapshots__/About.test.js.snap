// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutView shows the correct snapshot for the About component tree 1`] = `
<div
  className={
    Object {
      "data-css-1rdbgdp": "",
    }
  }
>
  <div
    className="css-jlylsu"
  >
    <p>
      About 
      <em>
        body
      </em>
    </p>
  </div>
</div>
`;
